sys/vm/vm_cheri_revoke.c

```diff
@@ -515,8 +546,8 @@ vm_cheri_revoke_object_at(const struct vm_cheri_revoke_cookie *crc, int flags,
 		/*
 		 * This is kind of awkward; the page is busy and it's
 		 * not clear by whom.  But whoever it is needs our
-		 * object lock to unbusy.  So handle this like the
-		 * map stepping forward.
+		 * object lock to unbusy.  (XXXMJ this is not generally
+		 * true.)  So handle this like the map stepping forward.
```

Comment:
Oh, I thought the busy lock could only change state while the object lock was held.  Is that only true for the unbusy->[sx]busy transitions and not the [sx]busy->unbusy direction?

Reply:
Right, it's possible to unbusy a page without holding the object lock. In older versions of FreeBSD the object might typically have been locked across unbusy, but there have been many changes in the past year or two to reduce the scope of the object lock, and even before that I believe unbusying without the object lock was permitted.